### ISSUE TITLE
False positives MD032 & MD007 on combinations of definition list and (un)ordered list

### ISSUE THREAD

**Issue by exavolt** [DavidAnson/markdownlint#318]
In VSCode:  
<img width="412" alt="Screen Shot 2020-08-15 at 11 54 47" src="https://user-images.githubusercontent.com/77626/90305549-39b57f00-deee-11ea-84a2-38d337ea162c.png">

Rendered by Hugo as expected:  
<img width="462" alt="Screen Shot 2020-08-15 at 11 55 12" src="https://user-images.githubusercontent.com/77626/90305552-44701400-deee-11ea-8df4-41dbd27b1a75.png">

The only way to combine a definition list with (un)ordered list is to put the first item at the same line as the colon (which translates to `<dd>`), which triggers MD032. In turn, it requires the subsequent items to be indented too, which triggers MD007.

The expectation is that it should be treated similarly to nested lists.

This won't trigger any of those rules:

```
- One
  - One of one
  - Two of one
```

**Comment by DavidAnson** [comment]
Definition lists are not part of the CommonMark specification and not currently supported. More context in this comment (the rest of that issue is a bit muddled): https://github.com/DavidAnson/markdownlint/issues/286#issuecomment-624454213
